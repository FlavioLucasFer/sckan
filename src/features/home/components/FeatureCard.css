#container {
	position: relative;
}

.feature-icon {
	position: absolute;
	right: 0;
	bottom: 0;

	font-size: 190px !important;
	color: #eee;
	cursor: default;
	
	transition: color 1s;
	animation: icon-animation-out ease-in-out;
}

@media screen and (max-width: 1440px) {
	.feature-icon {
		font-size: 170px !important;
	}
}

@media screen and (max-width: 1200px) {
	.feature-icon {
		font-size: 160px !important;
	}
}

@media screen and (max-width: 992px) {
	.feature-icon {
		font-size: 120px !important;
	}
}

@media screen and (max-width: 600px) {
	.feature-icon {
		font-size: 190px !important;
	}
}

@media screen and (max-width: px) {
	.feature-icon {
		font-size: 160px !important;
	}
}

#container:hover .feature-icon {
	color: #bdbdbd;
	transition: color 1s;
	animation: icon-animation-in ease-in-out;
}

@keyframes icon-animation-in {
	from {
		color: #eee;
	}
	to {
		color: #bdbdbd;
	}
}

@keyframes icon-animation-out {
	from {
		color: #bdbdbd;
	}
	to {
		color: #eee;
	}
}

.feature-card-title {
	font-size: 28px !important;
}

