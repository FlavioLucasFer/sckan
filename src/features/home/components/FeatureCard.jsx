import './FeatureCard.css';
import React from 'react';
import PropTypes from 'prop-types';

import Icon from 'core/components/Icon';
import Card from 'core/components/Card';

FeatureCard.propTypes = {
	featureIcon: PropTypes.string.isRequired,
	children: PropTypes.element,
};

function FeatureCard(props) {
	const {
		featureIcon,
		children,
		...rest
	} = props;

	return (
		<div id="container">
			<Card {...rest}
				className="s12"
				cardTitleClassName="feature-card-title"
				style={{ padding: 0 }} >
				{children}
				
				<Icon className="feature-icon">
					{featureIcon}
				</Icon>
			</Card>
		</div>
	);
}

export default FeatureCard;
