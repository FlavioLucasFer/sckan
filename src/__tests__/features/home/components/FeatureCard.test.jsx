import React from 'react';
import { shallow } from 'enzyme';

import FeatureCard from 'features/home/components/FeatureCard';

describe('Testing FeatureCard home feature component', () => {
	it('should render correctly', () => {
		const wrapper = shallow(
			<FeatureCard icon="business">
				<h2>testing feature card</h2>
			</FeatureCard>
		);

		expect(wrapper.debug()).toMatchSnapshot();

		wrapper.setProps({
			icon: 'add',
		});

		expect(wrapper.debug()).toMatchSnapshot();
	});
});