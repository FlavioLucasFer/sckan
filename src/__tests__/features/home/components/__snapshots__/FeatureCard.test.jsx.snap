// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing FeatureCard home feature component should render correctly 1`] = `
"<div id=\\"container\\">
  <ForwardRef(Card) icon=\\"business\\" className=\\"s12\\" cardTitleClassName=\\"feature-card-title\\" style={{...}}>
    <h2>
      testing feature card
    </h2>
    <Icon className=\\"feature-icon\\" />
  </ForwardRef(Card)>
</div>"
`;

exports[`Testing FeatureCard home feature component should render correctly 2`] = `
"<div id=\\"container\\">
  <ForwardRef(Card) icon=\\"add\\" className=\\"s12\\" cardTitleClassName=\\"feature-card-title\\" style={{...}}>
    <h2>
      testing feature card
    </h2>
    <Icon className=\\"feature-icon\\" />
  </ForwardRef(Card)>
</div>"
`;
